exonum/src/node/events.rs

```diff
@@ -68,29 +68,31 @@ impl NodeHandler {
             }
             ExternalMessage::Enable(value) => {
                 let s = if value { "enabled" } else { "disabled" };
-                if self.is_enabled == value {
+                if self.is_network_enabled == value {
                     info!("Node is already {}", s);
                 } else {
-                    self.is_enabled = value;
-                    self.api_state().set_enabled(value);
+                    self.is_network_enabled = value;
+                    self.api_state().set_network_enabled(value);
                     info!("The node is {} now", s);
-                    if self.is_enabled {
+                    if self.is_network_enabled {
+                        for address in self.state.connect_list().clone().peers.values() {
+                            self.connect(address);
+                        }
                         self.add_round_timeout();
+                    } else {
+                        for address in self.state.connections().clone().keys() {
+                            self.state.remove_peer_with_addr(address);
```

Comment:
Seems that if we remove all peers here we will not be able to enable network again.